lib/improver/tests/nowcasting/accumulation/test_Accumulation.py

```diff
@@ -168,97 +177,224 @@ def test_times(self):
+            plugin.process(self.cubes)
 
-        time_interval = 60
-        expected = [self.cubes[0:6], self.cubes[5:]]
+    def test_raises_exception_for_impossible_aggregation(self):
+        """Test function raises an exception when attempting to create an
+        accumulation_period that cannot be created from the input cubes."""
 
-        plugin = Accumulation(accumulation_period=300)
-        result = plugin._get_period_sets(time_interval, self.cubes)
+        plugin = Accumulation(accumulation_period=119)
+        msg = "The specified accumulation period "
 
-        for index, sublist in enumerate(result):
-            self.assertSequenceEqual(sublist, expected[index])
+        with self.assertRaisesRegex(ValueError, msg):
+            plugin._check_inputs(self.cubes)
 
-    def test_returns_all_cubes_if_period_unspecified(self):
-        """Test function returns a list containing the original cube list if
-        the accumulation_period is not set."""
 
-        time_interval = 60
-        plugin = Accumulation()
-        result = plugin._get_period_sets(time_interval, self.cubes)
-        self.assertSequenceEqual(result, [self.cubes])
+class Test__get_cube_subsets(rate_cube_set_up):
 
-    @ManageWarnings(record=True)
-    def test_raises_warning_for_unused_cubes(self, warning_list=None):
-        """Test function raises a warning when there are insufficient cubes to
-        complete the last period. In this test the accumulation period is 3
-        minutes, but the total span of rates cubes covers 10 minutes, resulting
-        in 3 complete periods and a final incomplete period that is not
-        returned. This test checks that a warning is raised to highlight that
-        there is an incomplete final period that is not returned."""
+    """Test the _get_cube_subsets method."""
 
-        time_interval = 60
-        warning_msg = (
+    def test_basic(self):
+        """Test that the subset of cubes that are within the accumulation
+        period are correctly identified. In this case, the subset of cubes
+        used for each accumulation period is expected to consist of 5 + 1
+        cubes, so the integral is 5."""
+        expected_cube_subset = self.cubes[:6]
+        upper_bound_fp, = self.cubes[5].coord("forecast_period").points
+        integral = 5
+        plugin = Accumulation(
+            accumulation_period=5*60,
+            forecast_periods=np.array([5])*60)
+        result = plugin._get_cube_subsets(
+            self.cubes, upper_bound_fp, integral)
+        self.assertEqual(expected_cube_subset, result)
+
+    @ManageWarnings(record=True)
+    def test_partial_period(self, warning_list=None):
+        """Test that if the forecast period of the upper bound cube is
+        not within the list of requested forecast periods, then the
+        subset of cubes returned is equal to None."""
+        expected_cube_subset = None
+        expected_warning_msg = (
```

Comment:
Is that what is actually being tested? You've forced integral to 0, so unless that the number of cubes in the subset is 1 (which we know from the otherwise identical test above, it is not) we get the warning. But this hasn't actually tested the iris extract method (as far as I can see?) that looks at the forecast periods.

Reply:
I think I've now updated this unit test, so that it's a bit more representative of the circumstance that we want to test.